### ISSUE TITLE
feature: Add support for subfolders

### ISSUE THREAD

**Issue by Zarkness** [LuaLink/LuaLink#53]
add support for subfolders and nested folders inside the scripts folder to organize scripts

**Comment by Saturn745** [comment]
I plan on adding something like this where you can have a folder that contains a main file and you can load scripts inside one another (with Lua's require function) when I finish moving the Lua API to use LuaJava instead of LuaJ. Unfortunately, I am away from home for a few months and won't be able to work on that for a little while. 

However, I would happily accept a PR that implements a simpler version of it, which just loads scripts from the scripts folder recursively for the time being.

**Comment by Zarkness** [comment]
okey, thx

**Comment by Zarkness** [comment]
one doubt that came to my mind now, do you only plan to have support for lua scripts? or do you plan to add other languages like kotlin, java, javascript, ruby or rust or any other language?

**Comment by Saturn745** [comment]
> one doubt that came to my mind now, do you only plan to have support for lua scripts? or do you plan to add other languages like kotlin, java, javascript, ruby or rust or any other language?  
>   
> I say this because I am looking for a plugin of this style but for kotlin so I force myself to learn it since I make the scripts in kotlin and then I can make my own plugins or help plugins like this one.

For this project, I only plan on supporting Lua, but I was considering making another project that adds scripting with Kotlin.

**Comment by Zarkness** [comment]
the kotlin one interests me a lot because one of the plugins I had in mind to do when I know kotlin was to make one with support for kotlin scripts in both .kts and .kt and gradually add other languages, but for now I don't know much about kotlin and as I'm more focused on configuring my server I leave aside kotlin and programming, so I was interested in one with support for kotlin.

**Comment by Saturn745** [comment]
> the kotlin one interests me a lot because one of the plugins I had in mind to do when I know kotlin was to make one with support for kotlin scripts in both .kts and .kt and gradually add other languages, but for now I don't know much about kotlin and as I'm more focused on configuring my server I leave aside kotlin and programming, so I was interested in one with support for kotlin.

Yeah, Kotlin scripting is still on my list of things to at least experiment with, so it may be something I do in the near future under a different project. 
Adding multiple languages to the same plugin, I feel, would get messy and bloated very quickly.

**Comment by Zarkness** [comment]
well I will be happy to try your plugin, in fact my idea to add support for different languages is that you need somehow like an addon to add support for that language.
